cmake/kokkos_corner_cases.cmake

```diff
@@ -1,4 +1,4 @@
-IF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang AND KOKKOS_ENABLE_OPENMP AND NOT KOKKOS_CLANG_IS_CRAY)
+IF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang AND KOKKOS_ENABLE_OPENMP AND NOT KOKKOS_CLANG_IS_CRAY AND NOT "x${CMAKE_CXX_SIMULATE_ID}" STREQUAL "xMSVC")
```

Comment:
```suggestion
IF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang AND KOKKOS_ENABLE_OPENMP AND NOT KOKKOS_CLANG_IS_CRAY AND NOT "${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
```

Reply:
Unfortunately without prepending 'x' it sometimes does not work. Random people on the internet have similar problems.

Reply:
Apparently even kitware does this inside of cmake: 
https://gitlab.kitware.com/cmake/cmake/commit/d44c0db0b2f8fa24b932b5c0d506fc8842baffd6

I guess just another ""feature"" of cmake ...